### PR TITLE
feat: support passing component with template to slots

### PR DESCRIPTION
**Pull request by xanf**
Pretty self-describing :)

Recover functionality which was available in v1 :)

### CODE REVIEW

File: docs/guide/advanced/slots.md [hunk 1]
```diff
@@ -134,8 +134,6 @@ test('layout full page layout', () => {
 })
 ```
 
-Note: passing a component using `{ template: '<div /> }` is not supported. Use a HTML string, render function, plain text, or an SFC.
```
Comment (afontcu) [inline docs/guide/advanced/slots.md:137]: What do you think about adding an example? Just adding a simple case above (lines 127 and 133) would do

**Review by afontcu — APPROVED** [review]
LGTM! Added a small suggestion for docs

**Review by lmiller1990 — APPROVED** [review]
Nice, can't believe we forgot this use case.

File: docs/guide/advanced/slots.md (hunk 1 above)
Comment (lmiller1990) [reply docs/guide/advanced/slots.md:137]: Does anyone remember why this was not supported? 🤔

Comment (xanf) [reply docs/guide/advanced/slots.md:137]: @afontcu makes sense, will do it ASAP! Also found a couple more places to clean up comments :)

**Comment by afontcu** [comment]
> Does anyone remember why this was not supported? 🤔

I can't remember of any good reason other than setting the initial structure of the stubbing mechanism was complex enough that we decided to go with the simple solutions. That one seemed harder than it was 😇

File: docs/guide/advanced/slots.md (hunk 1 above)
Comment (lmiller1990) [reply docs/guide/advanced/slots.md:137]: LMK when this is done, we can merge it up. Thanks!

Comment (xanf) [reply docs/guide/advanced/slots.md:137]: @lmiller1990 @afontcu Done :rocket: !

**Pull request merged**